### PR TITLE
Attempt at solution for the airports challenge

### PR DESCRIPTION
**Pull request by rahulrama**
This is my attempt at the solution for the airports challenge presented to us during Week 1 of the Makers Academy Course. 

'Pull early and often' - this solution will have multiple commits that will update the pull request automatically along the way, so as to check conformity with the conditions imposed by the Hound.

Meow...

### CODE REVIEW

File: lib/plane.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  def initialize(landed = false)
+    @landed = landed
+  end
+
+  def land
+    @landed = true
+  end
+
+  def take_off
+    @landed = false
```
Comment (danldb) [inline lib/plane.rb]: why use the `@` when you have an attr_accessor?

File: lib/plane.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  end
+
+  def docked?
+    landed
+  end
+
+  private
+  attr_accessor :landed
```
Comment (danldb) [inline lib/plane.rb]: This could be an attr_writter

**Comment by danldb** [comment]
:eyes:
A few of the errors in airport actually belonged in Plane (they purely related to the state of the plane).

Use DI for the weather.

**Comment by rahulrama** [comment]
Thanks for the feedback - i'll try and redo the exercise, and if not, atleast incorporate the changes into the current weeks'

**Pull request closed without merging**